src/openshift/cluster.ts

```diff
@@ -1042,6 +1042,44 @@ export class Cluster extends OpenShiftItem {
         return Cluster.tokenLogin(apiEndpointUrl, true, clipboard);
     }
 
+    static async loginUsingPipelineServiceAccountToken(server: string, proxy: string, username: string, accessToken: string ): Promise<string> {
+        const kcu = new k8s.KubeConfig();
+        const clusterProxy = {
+            name: 'sandbox-proxy',
+            server: proxy,
+        };
+        const user = {
+            name: 'sso-user',
+            token: accessToken,
+        };
+        const context = {
+            cluster: clusterProxy.name,
+            name: 'sandbox-proxy-context',
+            user: user.name,
+            namespace: `${username}-dev`,
+        };
+        kcu.addCluster(clusterProxy);
+        kcu.addUser(user)
+        kcu.addContext(context);
+        kcu.setCurrentContext(context.name);
+
+        const k8sApi = kcu.makeApiClient(k8s.CoreV1Api);
+        // const auth = new k8s.HttpBearerAuth();
+        // auth.accessToken = accessToken;
+        // k8sApi.setDefaultAuthentication(auth);
+
+        const serviceAccounts = await k8sApi.listNamespacedServiceAccount(`${username}-dev`);
+        const pipelineServiceAccount = serviceAccounts.body.items.find(serviceAccount => serviceAccount.metadata.name === 'pipeline');
+        if (!pipelineServiceAccount) {
+            return;
+        }
+        const secrets = await k8sApi.listNamespacedSecret(`${username}-dev`);
+        const pipelineTokenSecret = secrets.body.items.find((secret) => secret.metadata.name.startsWith('pipeline-token'));
```

Comment:
@dgolovin  After some months the `pipeline-token`  secret has disappeared and cannot be obtained anymore. Or is it only for my account?

F.i. my `secrets` consist of the following items:

- builder-dockercfg-ztqmw
- default-dockercfg-stxjj
- deployer-dockercfg-n4fgh
- model-serving-proxy-tls
- pipeline-dockercfg-mpct2

... and none of them has `*.data.token` property set. So, again, we can't use this login method for accessing the DevSandbox.

Reply:
It looks like we actually can create a token that can be used to authentication in DevSandbox...
If a `pipeline-token` doesn't exist we can do the following to create a new one:
```
            if (!pipelineTokenSecret) {
                const tokenRequest = new AuthenticationV1TokenRequest();
                tokenRequest.spec = {
                    audiences: ['https://kubernetes.default.svc'],
                    expirationSeconds: 600
                } as V1TokenRequestSpec;

                const pipelineTokenSecretResponse = await k8sApi.createNamespacedServiceAccountToken('pipeline', `${username}-dev`, tokenRequest);
                if (pipelineTokenSecretResponse?.body?.status?.token) {
                    return pipelineTokenSecretResponse.body.status.token;
                }
                return;
            }
```  
... and the token resulting in `pipelineTokenSecretResponse.body.status.token` (if exists) can be successfully used in `oc login --token="${pipelineTokenSecretResponse.body.status.token}"` 

However it doesn't look like the token created this way is stored in any service account or can be obtained from the namespace secrets, so it looks like a one-time-use time limited token. Not sure whether we can reuse it more than once and how many of such tokens can be created for a user.  
Also I haven't checked yet what happens if such a token expires (10 minutes is the minimum expiration time for a token) and what can be the maximum expiration time.